modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/GetObjectRelationshipEdgeCandidatesMVCResourceCommandTest.java

```diff
@@ -218,6 +218,37 @@ public void testGetObjectRelationships() throws Exception {
 			objectRelationshipBBB_AAAA.getObjectRelationshipId());
 	}
 
+	@Test
+	public void testPreventBidingWithinPublishedDefinitions() throws Exception {
+		ObjectDefinition objectDefinition1 =
+			ObjectDefinitionTestUtil.addCustomObjectDefinition(
+				RandomTestUtil.randomString(), _objectDefinitionLocalService);
+
+		ObjectDefinition objectDefinition2 =
+			ObjectDefinitionTestUtil.addCustomObjectDefinition(
+				RandomTestUtil.randomString(), _objectDefinitionLocalService);
+
+		_objectRelationshipLocalService.addObjectRelationship(
+			TestPropsValues.getUserId(),
+			objectDefinition1.getObjectDefinitionId(),
+			objectDefinition2.getObjectDefinitionId(), 0,
+			ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			StringUtil.randomId(),
+			ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		_objectDefinitionLocalService.publishCustomObjectDefinition(
+			TestPropsValues.getUserId(),
+			objectDefinition1.getObjectDefinitionId());
+
+		Assert.assertEquals(
+			_jsonFactory.createJSONArray(
+			).toString(),
+			_getObjectRelationshipEdgeCandidatesJSONArray(
+				2, objectDefinition2.getObjectDefinitionId()
+			).toString());
+	}
```

Comment:
Hey @thallescezar I've provided below a snippet of code demonstrating how I would write this test, I provided tests for other cases as well, could you please take a look and double check if these tests are fine?

Thank you!

```java
	@Test
	public void testIsEdgeCandidate() throws Exception {

		// Child object definition published

		ObjectDefinition childObjectDefinition1 =
			ObjectDefinitionTestUtil.addCustomObjectDefinition(
				_objectDefinitionLocalService);

		_objectDefinitionLocalService.publishCustomObjectDefinition(
			TestPropsValues.getUserId(),
			childObjectDefinition1.getObjectDefinitionId());

		ObjectDefinition parentObjectDefinition1 =
			ObjectDefinitionTestUtil.addCustomObjectDefinition(
				_objectDefinitionLocalService);

		ObjectRelationship objectRelationship1 =
			ObjectRelationshipTestUtil.addObjectRelationship(
				_objectRelationshipLocalService, parentObjectDefinition1,
				childObjectDefinition1);

		Assert.assertEquals(
			_jsonFactory.createJSONArray(
			).toString(),
			_getObjectRelationshipEdgeCandidatesJSONArray(
				0, childObjectDefinition1.getObjectDefinitionId()
			).toString());

		_objectRelationshipLocalService.deleteObjectRelationship(
			objectRelationship1.getObjectRelationshipId());

		// Many to Many object relationship

		ObjectDefinition objectDefinition1 =
			ObjectDefinitionTestUtil.addCustomObjectDefinition(
				_objectDefinitionLocalService);

		ObjectDefinition objectDefinition2 =
			ObjectDefinitionTestUtil.addCustomObjectDefinition(
				_objectDefinitionLocalService);

		ObjectRelationship objectRelationship2 =
			_objectRelationshipLocalService.addObjectRelationship(
				TestPropsValues.getUserId(),
				objectDefinition1.getObjectDefinitionId(),
				objectDefinition2.getObjectDefinitionId(), 0,
				ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
				StringUtil.randomId(), false,
				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);

		Assert.assertEquals(
			_jsonFactory.createJSONArray(
			).toString(),
			_getObjectRelationshipEdgeCandidatesJSONArray(
				0, objectDefinition2.getObjectDefinitionId()
			).toString());

		_objectRelationshipLocalService.deleteObjectRelationship(
			objectRelationship2.getObjectRelationshipId());

		// Parent object definition published

		ObjectDefinition childObjectDefinition2 =
			ObjectDefinitionTestUtil.addCustomObjectDefinition(
				_objectDefinitionLocalService);

		ObjectDefinition parentObjectDefinition2 =
			ObjectDefinitionTestUtil.addCustomObjectDefinition(
				_objectDefinitionLocalService);

		_objectDefinitionLocalService.publishCustomObjectDefinition(
			TestPropsValues.getUserId(),
			parentObjectDefinition2.getObjectDefinitionId());

		ObjectRelationship objectRelationship3 =
			ObjectRelationshipTestUtil.addObjectRelationship(
				_objectRelationshipLocalService, parentObjectDefinition2,
				childObjectDefinition2);

		Assert.assertEquals(
			_jsonFactory.createJSONArray(
			).toString(),
			_getObjectRelationshipEdgeCandidatesJSONArray(
				0, childObjectDefinition2.getObjectDefinitionId()
			).toString());

		_objectRelationshipLocalService.deleteObjectRelationship(
			objectRelationship3.getObjectRelationshipId());

		// Self object relationship

		ObjectDefinition objectDefinition3 =
			ObjectDefinitionTestUtil.addCustomObjectDefinition(
				_objectDefinitionLocalService);

		ObjectRelationship objectRelationship4 =
			ObjectRelationshipTestUtil.addObjectRelationship(
				_objectRelationshipLocalService, objectDefinition3,
				objectDefinition3);

		Assert.assertEquals(
			_jsonFactory.createJSONArray(
			).toString(),
			_getObjectRelationshipEdgeCandidatesJSONArray(
				0, childObjectDefinition1.getObjectDefinitionId()
			).toString());

		_objectRelationshipLocalService.deleteObjectRelationship(
			objectRelationship4.getObjectRelationshipId());
		
	}
```

Reply:
On "Many to Many" scenario you create a relationship with type ONE_TO_MANY. Is that correct?

Reply:
no, my fault, could you please replace to MANY_TO_MANY?

Thanks!